### DISCUSSION TITLE
WinForm - How to change the appearance of the context menu?

### DISCUSSION THREAD

**Discussion by MoeHiro** [cefsharp/CefSharp#3923] [General]
The default appearance of the context menu is not good-looking, and no way has been found to add icons. so is there a way to change it? Redraw it? Or can I use other controls instead, like ContextMenuStrip?

I tried to return true in RunContextMenu() and new a ContextMenuStrip, but it was not perfect.

**Comment by amaitland** [comment]
In future please use  [Discussions](https://github.com/cefsharp/CefSharp/discussions) to ask questions.

> I tried to return true in RunContextMenu() and new a ContextMenuStrip, but it was not perfect.

Yes you can implement your own custom menu. See https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Wpf/Handler/ContextMenuHandler.cs for a WPF example, should give you a rough idea.

If you require further assistance then please provide an example of your code.
